src/cargo.rs

```diff
@@ -432,6 +434,56 @@ pub fn rustdoc_flags(config: &Config) -> String {
     value
 }
 
+fn look_for_rustflags_in(path: &Path) -> Option<String> {
+    if let Ok(contents) = read_to_string(path) {
+        let value = contents.parse::<Value>().unwrap();
+
+        if let Some(build_value) = value.get("build") {
+            let build_table = build_value.as_table().unwrap();
+
+            if let Some(rustflags) = build_table.get("rustflags") {
+                let vec_of_flags: Vec<String> = rustflags
+                    .as_array()
+                    .unwrap()
+                    .into_iter()
+                    .map(|x| String::from(x.as_str().unwrap()))
```

Comment:
And one final unwrap to take care of

Reply:
Do you mean inside the map()?

Reply:
Yeah, I guess there might be an `as_str_lossy` but I'm not sure what the type is.

Reply:
I couldn't find a lot of info on as_str_lossy()? 

If I add a ?, the compiler says I dont have the correct return type:

                     .map(|x| String::from(x.as_str()?))
    |                          -----------------^^^^^^^^^^^-
    |                          |                |
    |                          |                cannot use the `?` operator in a closure that returns `std::string::String`
    |                          this function should return `Result` or `Option` to accept `?`

which kind of makes sense I guess but is there a something sneaky I can do here?

Reply:
ah okay. I looked at the toml docs properly.

So `.as_array()?` to remove that unwrap. And then for the Map you can do `.filter_map(|x| x.as_str()).map(|x| x.to_string())`

filter_map takes in a function that returns an option and discards the `None` values and unwraps the `Some(t)` to just the value inside. Then the map I put after that is to convert the `&str` to a `String`

Reply:
Ok got it - cheers Daniel.